lucet-runtime/lucet-runtime-internals/src/module/mock.rs

```diff
@@ -99,8 +101,40 @@ impl MockModuleBuilder {
         self
     }
 
-    pub fn with_export_func(mut self, sym: &[u8], func: *const extern "C" fn()) -> Self {
-        self.export_funcs.insert(sym.to_vec(), func);
+    fn record_sig(&mut self, sig: Signature) -> UniqueSignatureIndex {
+        let idx = self
```

Comment:
This might look familiar, which is because it's the same function as `declare_signature` in `ModuleInfo`. This greatly simplifies building up mock modules, instead of requiring us to track what signatures a mock module should have and their corresponding indices, so I think the copy is worthwhile.

Reply:
sgtm.